Sources/Controllers/MyPlaces/TracksViewController.swift

```diff
@@ -956,6 +1190,33 @@ class TracksViewController: OACompoundViewController, UITableViewDelegate, UITab
         }
     }
     
+    private func areAllItemsSelected() -> Bool {
+        guard let currentFolder = getTrackFolderByPath(currentFolderPath) else { return false }
+        let allDisplayedTracks = currentFolder.getAllTracks()
+        let allDisplayedFolders = currentFolder.subfolders.values
+        let allTracksSelected = allDisplayedTracks.allSatisfy { track in
+            if selectedTracks.contains(track) {
+                return true
+            }
+            
+            return selectedFolders.contains { folderName -> Bool in
+                let folderPath = currentFolder.path.appendingPathComponent(folderName)
+                let trackPath = track.gpxFilePath.deletingLastPathComponent()
+                return trackPath.hasPrefix(folderPath) || trackPath == folderPath
```

Comment:
|| trackPath == folderPath - Is this an extraneous condition?

Reply:
The condition trackPath == folderPath is necessary to ensure that tracks located directly in the specified folder, and not in any subfolders, are correctly identified.